src/typechecker.rs

```diff
@@ -3405,27 +3405,30 @@ pub fn typecheck_statement(
 
             (CheckedStatement::While(checked_cond, checked_block), error)
         }
-        ParsedStatement::Return(expr, span) => {
-            let (output, err) = typecheck_expression(
-                expr,
-                scope_id,
-                project,
-                safety_mode,
-                project
-                    .current_function_index
-                    .map(|i| project.functions[i].return_type_id),
-            );
-            error = error.or(err);
+        ParsedStatement::Return(expr, span) => match expr {
+            Some(e) => {
+                let (output, err) = typecheck_expression(
+                    e,
+                    scope_id,
+                    project,
+                    safety_mode,
+                    project
+                        .current_function_index
+                        .map(|i| project.functions[i].return_type_id),
+                );
+                error = error.or(err);
 
-            if project.inside_defer {
-                error = error.or(Some(JaktError::TypecheckError(
-                    "‘return’ is not allowed inside ‘defer’".to_string(),
-                    *span,
-                )));
-            }
+                if project.inside_defer {
+                    error = error.or(Some(JaktError::TypecheckError(
+                        "‘return’ is not allowed inside ‘defer’".to_string(),
+                        *span,
+                    )));
+                }
 
-            (CheckedStatement::Return(output), error)
-        }
+                (CheckedStatement::Return(Some(output)), error)
+            }
+            None => (CheckedStatement::Return(None), error),
```

Comment:
`return` should also not be allowed in `defer` as above?

Reply:
I'll add that check